### PR TITLE
feat(superagent-wrapper.js): add support for superagent-wrapper

### PR DESCRIPTION
**Pull request by ranrib**

### CODE REVIEW

**Comment by ten-lac** [comment]
Thank you @ranrib  for implementing this.

File: src/events/superagent-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+            const epsagonTraceId = httpHelpers.generateEpsagonTraceId();
+            // Inject header to support tracing over HTTP requests
+            if ((process.env.EPSAGON_DISABLE_HTTP_TRACE_ID || '').toUpperCase() !== 'TRUE') {
+                req.header[EPSAGON_HEADER] = epsagonTraceId;
+            }
+
+            eventInterface.addToMetadata(httpEvent,
+                {
+                    url: req.url,
```
Comment (haimrait) [inline src/events/superagent-wrapper.js:42]: what with the body of the requests?

File: src/events/superagent-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+                    http_trace_id: epsagonTraceId,
+                }, {
+                    request_headers: req.header,
```
Comment (haimrait) [inline src/events/superagent-wrapper.js:45]: its req.header not headers?

Comment (ranrib) [reply src/events/superagent-wrapper.js:45]: yes 🤦

File: src/events/superagent-wrapper.js (hunk 1 above)
Comment (ranrib) [reply src/events/superagent-wrapper.js:42]: Unfortunately, it's not reachable from this position (function) - it only comes as an async uncorrelated event afterwards, so we can't match it.

**Pull request merged**